src/index.ts

```diff
@@ -19,103 +19,42 @@
  * SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
  */
 
-// Consider redoing this file as it contains code outside of the dependency injection framework
-import ISolidSession from "./solidSession/ISolidSession";
 import ILoginInputOptions from "./ILoginInputOptions";
-import AuthFetcher from "./AuthFetcher";
-import getAuthFetcherWithDependencies from "./dependencies";
-import IStorage from "./localStorage/IStorage";
+import Session from "./Session";
 
-export interface ICustomAuthFetcherOptions {
-  storage?: IStorage;
-  doNotAutoHandleRedirect?: boolean;
-}
-
-let redirectHandlerPromise: Promise<void>;
-let redirectHandlerPromiseIsResolved = false;
-export async function customAuthFetcher(
-  options?: ICustomAuthFetcherOptions
-): Promise<AuthFetcher> {
-  const authFetcher = getAuthFetcherWithDependencies({
-    storage: options?.storage
-  });
-  if (!options?.doNotAutoHandleRedirect) {
-    if (!redirectHandlerPromise) {
-      redirectHandlerPromise = authFetcher.automaticallyHandleRedirect();
-      await redirectHandlerPromise;
-      redirectHandlerPromiseIsResolved = true;
-    } else if (!redirectHandlerPromiseIsResolved) {
-      await redirectHandlerPromise;
-    }
-  }
-  return authFetcher;
-}
-
-let globalAuthFetcher: AuthFetcher;
-async function getGlobalAuthFetcher(
-  options?: ICustomAuthFetcherOptions
-): Promise<AuthFetcher> {
-  if (globalAuthFetcher) {
-    return globalAuthFetcher;
-  }
-  globalAuthFetcher = await customAuthFetcher(options);
-  return globalAuthFetcher;
-}
-
-export async function login(
-  options: ILoginInputOptions
-): Promise<ISolidSession> {
-  const authFetcher = await getGlobalAuthFetcher();
-  return authFetcher.login(options);
+export async function login(options: ILoginInputOptions): Promise<void> {
+  throw new Error("Not implemented.");
```

Comment:
Same niggle as above - I'd just include the name of the not implemented method in the message string.